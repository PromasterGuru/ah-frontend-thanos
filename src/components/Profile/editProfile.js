import React from 'react';
import './profile.scss';
import PropTypes from 'prop-types';

const formElements = [
  {
    inputType: 'text',
    elementId: 'first_name',
    placeholder: 'First Name',
  },
  {
    inputType: 'text',
    elementId: 'last_name',
    placeholder: 'Last Name',
  },
];

const FormInput = ({ el, onChange }) => {
  const { inputType, elementId, placeholder } = el;
  return (

    <div className="form-group">
      <input
        type={inputType}
        placeholder={placeholder}
        className="form-control"
        id={elementId}
        name={elementId}
        onChange={onChange}
      />
    </div>
  );
};

FormInput.propTypes = {
  el: PropTypes.shape({
    inputType: PropTypes.string.isRequired,
    elementId: PropTypes.string.isRequired,
    placeholder: PropTypes.string.isRequired,
  }).isRequired,
  onChange: PropTypes.func.isRequired,
};

const EditProfile = ({ onSubmit, onChange, onClick }) => (

  <div className="container">
    <h1 className="title">Edit Profile</h1>
    <hr />
    <div className="row">
      <div className="col-md-3">
        <div className="text-center">
          <img src="https://scontent-jnb1-1.xx.fbcdn.net/v/t1.0-9/46369806_2116095768452511_152387389577232384_n.jpg?_nc_cat=106&_nc_ht=scontent-jnb1-1.xx&oh=6a5c22acd1b1a9e9058cf07f10c8426f&oe=5CAA0106" className="img-fluid" alt="avatar" />
          <h6>Upload a different photo...</h6>
          <input type="file" className="form-control" onInput={onClick} />
          <div className="col-lg-8 mt-2">
            <button type="submit" className="btn button">Save Changes</button>
          </div>
        </div>

      </div>

      <div className="col-md-9 personal-info">
        <h3 className="pl-3">Personal info</h3>

        <form className="form-horizontal" onSubmit={onSubmit}>
          {formElements.map(el => <FormInput onChange={onChange} el={el} key={el.elementId} />)}
          <textarea
            className="form-control"
            id="exampleFormControlTextarea1"
            rows="3"
            name="bio"
            onChange={onChange}
            placeholder="Tell us something about yourself."
          />
          <div className="col-lg-8">
            <button type="submit" className="btn  btn-outline-primary mt-4 button" onClick={onSubmit}>Save Changes</button>
            <span className="mr-2" />
            <button type="submit" className="btn btn-outline-danger mt-4">Cancel</button>
          </div>
        </form>
      </div>

    </div>
  </div>

);
EditProfile.propTypes = {
  onChange: PropTypes.func,
  onSubmit: PropTypes.func,
  onClick: PropTypes.func,
};
EditProfile.defaultProps = {
  onChange: true,
  onSubmit: true,
  onClick: true,
};

export default EditProfile;
