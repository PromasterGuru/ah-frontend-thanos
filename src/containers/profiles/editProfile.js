import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import EditProfile from '../../components/Profile/editProfile';
import { editProfileAction } from '../../actions/profileActions';
import { editProfile, uploadImage } from '../../actions/actionCreators';

export class EditProfilePage extends Component {
  handleUpdateFields = (event) => {
    const { name, value } = event.target;
    const { getUserInputs } = this.props;
    getUserInputs({ [name]: value });
  };

  handleSubmit = (event) => {
    event.preventDefault();
    const { editProfileDispatch } = this.props;
    editProfileDispatch();
  };

  handleUpload = () => {
    const widget = window.cloudinary.openUploadWidget({
      cloud_name: process.env.REACT_APP_CLOUD_NAME,
      upload_preset: process.env.REACT_APP_CLOUDINARY_UPLOAD_PRESET,
      cropping: true,
      folder: 'widgetdocs',
      sources: ['local', 'url', 'camera', 'facebook', 'dropbox', 'search', 'instagram'],
    }, (error, result) => {
      if (result.event === 'success') {
        const { editProfileDispatch } = this.props;
        this.dispatch(uploadImage({ image: result.info.secure_url }));
        editProfileDispatch();
        widget.close();
      }
    });
    widget.open();
  };

  render() {
    return (
      <EditProfile
        onChange={this.handleUpdateFields}
        onSubmit={this.handleSubmit}
        onClick={this.handleUpload}
      />
    );
  }
}

EditProfile.propTypes = {
  editProfileDispatch: PropTypes.func.isRequired,
  getUserInputs: PropTypes.func.isRequired,
  freshProfile: PropTypes.shape({}).isRequired,
};

const mapStateToProps = ({ profileReducer }) => (
  { freshProfile: profileReducer.freshProfile }
);
export const mapDispatchToProps = dispatch => (
  {
    editProfileDispatch: profile => dispatch(editProfileAction(profile)),
    getUserInputs: obj => dispatch(editProfile(obj)),
  }
);
export default connect(mapStateToProps, mapDispatchToProps)(EditProfilePage);
