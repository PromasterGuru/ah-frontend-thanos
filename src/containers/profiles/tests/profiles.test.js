import React from 'react';
import Enzyme, { mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { Provider } from 'react-redux';
import { EditProfilePage } from '../editProfile';
import EditProfile from '../../../components/Profile/editProfile';

Enzyme.configure({ adapter: new Adapter() });
const storeFake = state => ({
  default: jest.fn(),
  subscribe: jest.fn(),
  dispatch: jest.fn(),
  getState: () => state,
});

describe('profile container', () => {
  let wrapper;
  let component;
  let container;

  beforeEach(() => {
    jest.resetAllMocks();
    const store = storeFake({ profile: { username: 'test' } });

    wrapper = mount(
      <Provider store={store}>
        <EditProfilePage />
      </Provider>,
    );

    container = wrapper.find(EditProfilePage);
    component = wrapper.find(EditProfile);
  });

  it('should render both the container and the component ', () => {
    expect(container.length).toBeTruthy();
    expect(component.length).toBeTruthy();
  });
  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
