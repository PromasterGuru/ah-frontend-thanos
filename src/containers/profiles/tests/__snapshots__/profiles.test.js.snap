// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`profile container should render correctly 1`] = `
<Provider
  store={
    Object {
      "default": [MockFunction],
      "dispatch": [MockFunction],
      "getState": [Function],
      "subscribe": [MockFunction],
    }
  }
>
  <EditProfilePage>
    <EditProfile
      onChange={[Function]}
      onClick={[Function]}
      onSubmit={[Function]}
    >
      <div
        className="container"
      >
        <h1
          className="title"
        >
          Edit Profile
        </h1>
        <hr />
        <div
          className="row"
        >
          <div
            className="col-md-3"
          >
            <div
              className="text-center"
            >
              <img
                alt="avatar"
                className="img-fluid"
                src="https://scontent-jnb1-1.xx.fbcdn.net/v/t1.0-9/46369806_2116095768452511_152387389577232384_n.jpg?_nc_cat=106&_nc_ht=scontent-jnb1-1.xx&oh=6a5c22acd1b1a9e9058cf07f10c8426f&oe=5CAA0106"
              />
              <h6>
                Upload a different photo...
              </h6>
              <input
                className="form-control"
                onInput={[Function]}
                type="file"
              />
              <div
                className="col-lg-8 mt-2"
              >
                <button
                  className="btn button"
                  type="submit"
                >
                  Save Changes
                </button>
              </div>
            </div>
          </div>
          <div
            className="col-md-9 personal-info"
          >
            <h3
              className="pl-3"
            >
              Personal info
            </h3>
            <form
              className="form-horizontal"
              onSubmit={[Function]}
            >
              <FormInput
                el={
                  Object {
                    "elementId": "first_name",
                    "inputType": "text",
                    "placeholder": "First Name",
                  }
                }
                key="first_name"
                onChange={[Function]}
              >
                <div
                  className="form-group"
                >
                  <input
                    className="form-control"
                    id="first_name"
                    name="first_name"
                    onChange={[Function]}
                    placeholder="First Name"
                    type="text"
                  />
                </div>
              </FormInput>
              <FormInput
                el={
                  Object {
                    "elementId": "last_name",
                    "inputType": "text",
                    "placeholder": "Last Name",
                  }
                }
                key="last_name"
                onChange={[Function]}
              >
                <div
                  className="form-group"
                >
                  <input
                    className="form-control"
                    id="last_name"
                    name="last_name"
                    onChange={[Function]}
                    placeholder="Last Name"
                    type="text"
                  />
                </div>
              </FormInput>
              <textarea
                className="form-control"
                id="exampleFormControlTextarea1"
                name="bio"
                onChange={[Function]}
                placeholder="Tell us something about yourself."
                rows="3"
              />
              <div
                className="col-lg-8"
              >
                <button
                  className="btn  btn-outline-primary mt-4 button"
                  onClick={[Function]}
                  type="submit"
                >
                  Save Changes
                </button>
                <span
                  className="mr-2"
                />
                <button
                  className="btn btn-outline-danger mt-4"
                  type="submit"
                >
                  Cancel
                </button>
              </div>
            </form>
          </div>
        </div>
      </div>
    </EditProfile>
  </EditProfilePage>
</Provider>
`;
